pkg/matcher/README.md

```diff
@@ -52,16 +69,30 @@ The pattern syntax is:
         '\\' c      matches character c
         lo '-' hi   matches character c for lo <= c <= hi
 ```
+e.g 
+ - `* ?` will match any string that is a single character. 
+ - `'?a'` will match any 2 character string that starts with any character and the second character is `a`, like `ba` but not `bb` or `bba`. 
+ - `'[^abc]'` will match any character that is NOT a,b,c. `'[abc]'` will match only a, b, c.
+ - `'*[a-d]'` will match any string (`*`) that ends with a character that is between `a` and `d` (i.e `a,b,c,d`).
+
+
       
 ### Regexp matcher
 The regexp matcher reports whether the given value matches the RegExp pattern ( use regexp.Match ).
 
 The RegExp syntax is described at https://golang.org/pkg/regexp/syntax/.
 
+We know that Regular Expressions are hard, but they are powerfull. If you want to learn more about them, you can visit [RegexOne](https://regexone.com/).
+
 ### Simple patterns matcher
 The simple patterns matcher reports whether the given value matches the simple patterns.
 
-The simple patterns is a custom format used in netdata, it's syntax is described at https://docs.netdata.cloud/libnetdata/simple_pattern/.
+Simple patterns are a space separated list of words, that can have `*` as a wildcard. Each world may use any number of `*`. Simple patterns allow negative matches by prefixing a word with `!`.
+
+So, pattern = `!*bad* *` will match anything, except all those that contain the word bad.
+
+Simple patterns are quite powerful: pattern = `*foobar* !foo* !*bar *` matches everything containing foobar, except strings that start with foo or end with bar.
```

Comment:
This copied from simple patterns readme, i think we shouldnt do it and just provide a link. Some example would be good tho!

Reply:
I think we should provide all the necessary information in the readme, instead of redirecting them to our docs, we add useless friction.

Reply:
I dont think it is useless, the up-to-date description is netdata simple-patterns readme. Same goes for regex - we dont want to describe regex syntax when there is official page with the syntax

Reply:
Lets add it as an `Examples` section. Same as you did in `glob` matcher section.

```
Examples:
- `!*bad* *` will match anything, except all those that contain the word bad.
- ...
```

Reply:
Ok i have a second thought and i think you are right, we dont need the link to the netdata simple_patterns at all because go.d simple patterns has more features (for example netdata doesnt handle `[a-z]`).

So lets get Examples section here and it is ok

Reply:
simple patterns is basically a space separated list of glob patterns. Since it is a list we try them in turn and the result is the first one that mathes. In addition a glob pattern can be negated (means if it mathces it returns `false`)

Reply:
I think we should at least link to the simple patterns doc as a reference to that source of truth, even if they are slightly different implementations. Overall I think we need to increase visibility on the simple patterns syntax, as it's something that I've seen a few users stumble over since it's not a "standard" like regex.

Reply:
It is not really source of truth. It behaves same as netdata simple patterns in general:

- list of patterns
- execute them in turn
- first one that matches returns

pattern syntax itself is different, in addition to `*` go.d simple patterns support `?` `[...]` `[^...]

That is why i started to think that we dont need it 🤔